edk2toolext/tests/test_web_dependency.py

```diff
@@ -81,6 +94,20 @@ def test_fail_with_bad_url(self):
             ext_dep.fetch()
             self.fail("should have thrown an Exception")
 
+    # try to download a single file from github
+    def test_single_file(self):
+        ext_dep_file_path = os.path.join(test_dir, "good_ext_dep.json")
+        with open(ext_dep_file_path, "w+") as ext_dep_file:
+            ext_dep_file.write(single_file_json_file)
+
+        ext_dep_descriptor = EDF.ExternDepDescriptor(ext_dep_file_path).descriptor_contents
+        ext_dep = WebDependency(ext_dep_descriptor)
+        ext_dep.fetch()
+
+        file_path = os.path.join(test_dir, "test_extdep", "test.txt")
```

Comment:
would it be better to get test_extdep and test.txt from the "single_file_json_file" string?

Reply:
yeah but since it's a string, we don't have a good way of reading is as yaml. I'll fix it

Reply:
Fixed